homeassistant/components/home_connect/light.py

```diff
@@ -0,0 +1,209 @@
+            entities += entity_list
+        return entities
+
+    async_add_entities(await hass.async_add_executor_job(get_entities), True)
+
+
+class HomeConnectLight(HomeConnectEntity, LightEntity):
+    """Light for Home Connect."""
+
+    def __init__(self, device, desc, ambient):
+        """Initialize the entity."""
+        super().__init__(device, desc)
+        self._state = None
+        self._brightness = None
+        self._hs_color = None
+        self._ambient = ambient
+        if self._ambient:
+            self._brightnesskey = BSH_AMBIENTLIGHTBRIGHTNESS
+            self._key = BSH_AMBIENTLIGHTENABLED
+            self._customcolorkey = BSH_AMBIENTLIGHTCUSTOMCOLOR
+            self._colorkey = BSH_AMBIENTLIGHTCOLOR
+        else:
+            self._brightnesskey = COOKING_LIGHTINGBRIGHTNESS
+            self._key = COOKING_LIGHTING
+            self._customcolorkey = None
+            self._colorkey = None
+
+    @property
+    def is_on(self):
+        """Return true if the light is on."""
+        return bool(self._state)
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def hs_color(self):
+        """Return the color property."""
+        return self._hs_color
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        if self._ambient:
+            return SUPPORT_BRIGHTNESS | SUPPORT_COLOR
+        return SUPPORT_BRIGHTNESS
+
+    async def async_turn_on(self, **kwargs):
+        """Switch the light on, change brightness, change color."""
+        if self._ambient:
+            if ATTR_BRIGHTNESS in kwargs or ATTR_HS_COLOR in kwargs:
+                try:
+                    await self.hass.async_add_executor_job(
+                        self.device.appliance.set_setting,
+                        self._colorkey,
+                        BSH_AMBIENTLIGHTCOLOR_CUSTOMCOLOR,
+                    )
+                except HomeConnectError as err:
+                    _LOGGER.error("Error while trying selecting customcolor: %s", err)
+                if self._brightness is not None:
+                    brightness = 10 + ceil(self._brightness / 255 * 90)
+                    if ATTR_BRIGHTNESS in kwargs:
+                        brightness = 10 + ceil(kwargs[ATTR_BRIGHTNESS] / 255 * 90)
+
+                    hs_color = kwargs.get(ATTR_HS_COLOR, default=self._hs_color)
+
+                    if hs_color is not None:
+                        rgb = color_util.color_hsv_to_RGB(*hs_color, brightness)
+                        hex_val = color_util.color_rgb_to_hex(rgb[0], rgb[1], rgb[2])
+                        try:
+                            await self.hass.async_add_executor_job(
+                                self.device.appliance.set_setting,
+                                self._customcolorkey,
+                                "#" + hex_val,
```

Comment:
Please use f-strings instead of string concatenation.

Reply:
Done